### PR TITLE
feat: support google login on desktop

### PR DESCRIPTION
**Pull request by pengx17**
Thanks to the help of @darkskygit, it turns out that we don't need any hacks within Electron by creating a hidden window and handling login inside it. Now, we have a much easier solution:

- Instead of directly calling the signin function when signing in is initiated via Electron, we can pop up the login window in the browser and initiate signIn there.
- Once Google signin is finished, it can redirect to the open-app route, fetch the JWT token, and send it to the desktop app.
- The desktop app can then accept the token and set it to the browser window using cookie.

It seems like we don't need to change the electron origin after all. I will go back and fix email login as well later on.

### CODE REVIEW

File: apps/core/src/pages/desktop-signin.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { signIn, type SignInResponse } from 'next-auth/react';
+import { useEffect, useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { z } from 'zod';
+
+const supportedProvider = z.enum(['google']);
+
+let wip = false;
```
Comment (himself65) [inline apps/core/src/pages/desktop-signin.tsx]: use a atom

**Review by himself65 — CHANGES_REQUESTED** [review]

File: apps/core/src/pages/desktop-signin.tsx (hunk 1 above)
Comment (himself65) [reply apps/core/src/pages/desktop-signin.tsx]: yep

Comment (himself65) [reply apps/core/src/pages/desktop-signin.tsx]: I believe we will add e2e test on this. Please make sure the UI is pure

**Comment by pengx17** [comment]
related https://github.com/toeverything/AFFiNE/issues/3944

**Review by darkskygit — APPROVED** [review]

**Review by darkskygit — APPROVED** [review]

**Review by himself65 — APPROVED** [review]

**Pull request merged**